pkg/sources/gitlab/gitlab.go

```diff
@@ -81,6 +84,57 @@ func (s *Source) JobID() sources.JobID {
 	return s.jobID
 }
 
+// filteredRepoCache is a wrapper around cache.Cache that filters out repos
+// based on include and exclude globs.
+type filteredRepoCache struct {
+	cache.Cache[string]
+	include, exclude []glob.Glob
+}
+
+func (s *Source) newFilteredRepoCache(ctx context.Context, c cache.Cache[string], include, exclude []string) *filteredRepoCache {
+	includeGlobs := make([]glob.Glob, 0, len(include))
+	excludeGlobs := make([]glob.Glob, 0, len(exclude))
+	for _, ig := range include {
+		g, err := glob.Compile(ig)
+		if err != nil {
+			ctx.Logger().V(1).Info("invalid include glob", "include_value", ig, "err", err)
```

Comment:
If possible, I definitely think that glob compilation errors should be a fatal error. At the very least we should log it with `Error` instead of `Info`.

Reply:
Reverted the code back to support compilation code. infact, now compilation will be done at `validation`, `enumeration` or `chunk` func like it was done before.